pytorch_lightning/profiler/profilers.py

```diff
@@ -543,3 +549,252 @@ def __del__(self):
+                        f"Found schedule: {schedule}. "
+                        "Schedule should be a callable returning `torch.profiler.ProfilerAction`. "
+                    )
+
+            if isinstance(self.sort_by_key, str) and self.sort_by_key not in self.AVAILABLE_SORT_KEYS:
+                raise MisconfigurationException(
+                    f"Found sort_by_key: {self.sort_by_key}. Should be within {self.AVAILABLE_SORT_KEYS}. "
+                )
+
+            self.output_filename = output_filename
+            self.enabled = enabled
+            self.use_cpu = use_cpu
+            self.use_cuda = use_cuda
+            self.record_functions = record_functions or self.RECORD_FUNCTIONS
+            self.record_functions_managers = {}
+            self.activities = [ProfilerActivity.CPU] * use_cpu + [ProfilerActivity.CUDA] * use_cuda
+            self.schedule = ScheduleWrapper(schedule) if schedule is not None else schedule
+            self.record_shapes = record_shapes
+            self.row_limit = row_limit
+            self.export_to_tensorboard = export_to_tensorboard
+            self.export_to_flame_graph = export_to_flame_graph
+            self.profile_memory = profile_memory
+            self.with_stack = True if export_to_flame_graph else with_stack
+            self.with_flops = with_flops
+            self.local_rank = local_rank
+            self.path_to_export_trace = path_to_export_trace
+            self.group_by_input_shapes = group_by_input_shapes
+            # currently Lightning handles this part for the user
+            # Open an issue with the specified use case.
+            self.on_trace_ready = on_trace_ready
+            self.udf_on_trace_ready = True if self.on_trace_ready is not None else False
+
+            self.context_names = {}
+            self.running_stack = []
+            self.profiler = None
+
+            self.output_fname = output_filename
+            self.output_file = None
+            if local_rank is not None:
+                super().on_train_start(local_rank=local_rank)
+                self.on_train_start = super(LegacyPyTorchProfiler, self).on_train_start
+
+        def summary(self) -> str:
+            recorded_stats = {}
+            output_string = ''
+            local_rank = '0' if self.local_rank is None else self.local_rank
+
+            if not self.enabled:
+                return output_string
+
+            self.profiler.__exit__(None, None, None)
+
+            data = self.profiler.events().key_averages(group_by_input_shapes=self.group_by_input_shapes)
+            table = data.table(sort_by=self.sort_by_key, row_limit=self.row_limit)
+            recorded_stats[self.START_ACTION] = table
+
+            # log to standard out
+            output_string = f"{os.linesep}Profiler Report{os.linesep}"
+            for action, stats in recorded_stats.items():
+                output_string += (f"{os.linesep}Profile stats for: {action} rank: {local_rank} {os.linesep}{stats}")
```

Comment:
Nit: Maybe assign `os.linesep` to a variable. Repetitive use.

Reply:
Done !